Source/APIClient/APIClient.swift

```diff
@@ -2,16 +2,32 @@ import Alamofire
 
 final class APIClient {
 
+    enum HeaderType {
+        case chainpointJson
+        
+        func value() -> HTTPHeaders {
+            switch self {
+            case .chainpointJson:
+                return [
+                    "Content-Type":"application/json",
+                    "Accept": "application/vnd.chainpoint.ld+json"
+                ]
+            }
+        }
+    }
+    
     enum Error: Swift.Error {
         case missingResponseValue
     }
     // TODO: Daniel Metzing - write tests
 
-    func execute(request: BlockchainRequest, completion: @escaping (Result<APIResponse>) -> Void ) {
+    func execute(request: BlockchainRequest, headers: HeaderType? = nil, completion: @escaping (Result<APIResponse>) -> Void ) {
+        
         Alamofire.request(request.url,
                           method: request.httpMethod,
                           parameters: request.parameters,
-                          encoding: request.encoding)
+                          encoding: request.encoding,
+                          headers: headers?.value())
```

Comment:
If we add to the request, here we could just add `request.headers`. The requests would be still encapsulating all the parameters, and the `APIClient` could still stay very generic.